components/gantry/oneaxis/oneaxis.go

```diff
@@ -172,67 +172,50 @@ func (g *oneAxis) home(ctx context.Context, np int) error {
 	defer done()
 	// Mapping one limit switch motor0->limsw0, motor1 ->limsw1, motor 2 -> limsw2
 	// Mapping two limit switch motor0->limSw0,limSw1; motor1->limSw2,limSw3; motor2->limSw4,limSw5
-	switch np {
-	// An axis with one limit switch will go till it hits the limit switch, encode that position as the
-	// zero position of the oneaxis, and adds a second position limit based on the steps per length.
-	case 1:
-		if err := g.homeOneLimSwitch(ctx); err != nil {
-			return err
-		}
-	// An axis with two limit switches will go till it hits the first limit switch, encode that position as the
-	// zero position of the oneaxis, then go till it hits the second limit switch, then encode that position as the
-	// at-length position of the oneaxis.
-	case 2:
-		if err := g.homeTwoLimSwitch(ctx); err != nil {
+
+	if np > 0 {
```

Comment:
might be more readable to keep the case structure actually now that I think about it, and make it one big `Home` function that takes in the number of limit switch pins.

Reply:
so all the encoder logic in the same function as well?